### DISCUSSION TITLE
Extend setting base transform to other asset methods

### DISCUSSION THREAD

**Discussion by nixondesigndev** [craftcms/cms#8861] [Ideas]
### Description

This is a request to add support for specifying a base transform (https://github.com/craftcms/cms/issues/5853#issuecomment-652692013) to the `getSrcSet()` method in a similar way to `getUrl()`.

```
{% set srcsetSizes = [ '1.5x', '2x', '3x' ] %}
{% set baseTransform = { width: 300, height: 300 } %}

{% do asset.setTransform(baseTransform) %}

{{ tag('img', {
  src: asset.getUrl({ transform: baseTransform, format: 'webp' }),
  srcset: asset.getSrcset(srcsetSizes, { format: 'webp' }),
}) }}
```

Alternatively to align `getSrcset` with `getUrl`, `getSrcset` accepts an object as the first param?

```
getSrcset({ sizes: srcsetSizes, format: 'webp' })
```

Could/should this behaviour then be extended to `getImg`? In the case of overriding the format and using the `source` tag I don't this being beneficial, but a developer may want to output the same asset multiple times with different crops etc.

**Comment by brandonkelly** [comment]
That `$transformOverrideParameters` argument was removed in c4696c68064238a32d5bf53bd9249abbdd7c8c63, so just updated your issue.
